src/document.c

```diff
@@ -328,44 +317,8 @@ void AddDocumentCtx_Submit(RSAddDocumentCtx *aCtx, RedisSearchCtx *sctx, uint32_
   // We actually modify (!) the strings in the document, so we always require
   // ownership
   Document_MakeStringsOwner(aCtx->doc);
-
-  if (AddDocumentCtx_IsBlockable(aCtx)) {
-    aCtx->client.bc = RedisModule_BlockClient(sctx->redisCtx, replyCallback, NULL, NULL, 0);
-  } else {
-    aCtx->client.sctx = sctx;
-  }
-
-  RS_LOG_ASSERT(aCtx->client.bc, "No blocked client");
-
-  bool concurrentSearch = false;
-  if (AddDocumentCtx_IsBlockable(aCtx)) {
-    size_t totalSize = 0;
-    for (size_t ii = 0; ii < aCtx->doc->numFields; ++ii) {
-      const DocumentField *ff = aCtx->doc->fields + ii;
-      if ((ff->indexAs & (INDEXFLD_T_FULLTEXT | INDEXFLD_T_TAG))) {
-        // TODO: GEOMETRY - handle geometry fields?
-        size_t n;
-        if (ff->unionType == FLD_VAR_T_CSTR || ff->unionType == FLD_VAR_T_RMS) {
-          DocumentField_GetValueCStr(&aCtx->doc->fields[ii], &n);
-          totalSize += n;
-        } else if (ff->unionType == FLD_VAR_T_ARRAY) {
-          for (size_t jj = 0; jj < ff->arrayLen; ++jj) {
-            DocumentField_GetArrayValueCStr(&aCtx->doc->fields[ii], &n, jj);
-            totalSize += n;
-          }
-        }
-      }
-    }
-    concurrentSearch = (totalSize >= SELF_EXEC_THRESHOLD);
-  }
-
-  if (!concurrentSearch) {
-    Document_AddToIndexes(aCtx, sctx);
-  } else {
-    // Deprecated and broken - should pass `DocumentAddCtx` and not `RSAddDocumentCtx`
-    // also, we have to pass a weak ref to the spec, and handle it in the callback.
-    ConcurrentSearch_ThreadPoolRun(threadCallback, aCtx, CONCURRENT_POOL_INDEX);
-  }
+  aCtx->client.sctx = sctx;
```

Comment:
I guess we do not need the `aCtx->client.bc` anymore right? Did you deleted it?

Reply:
No. Good call